pages/common/diff.md

```diff
@@ -9,20 +9,24 @@
 
 - Compare files, ignoring white spaces:
 
-`diff -w {{old_file}} {{new_file}}`
+`diff --ignore-all-space {{old_file}} {{new_file}}`
 
 - Compare files, showing the differences side by side:
 
-`diff -y {{old_file}} {{new_file}}`
+`diff --side-by-side {{old_file}} {{new_file}}`
 
 - Compare files, showing the differences in unified format (as used by `git diff`):
 
-`diff -u {{old_file}} {{new_file}}`
+`diff --unified {{old_file}} {{new_file}}`
 
 - Compare directories recursively (shows names for differing files/directories as well as changes made to files):
 
-`diff -r {{old_directory}} {{new_directory}}`
+`diff --recursive {{old_directory}} {{new_directory}}`
 
 - Compare directories, only showing the names of files that differ:
 
-`diff -rq {{old_directory}} {{new_directory}}`
+`diff --recursive --brief {{old_directory}} {{new_directory}}`
+
+- Create patch file from the differences of two directories or files:
+
+`diff --text --unified --new-file {{old_file_or_directory}} {{new_file_or_directory}} > {{diff.patch}}`
```

Comment:
Sorry for the delay. Here's my last suggestion:
* Since we left out `--recoursive` we have to remove `directory`
* I think we should also leave out `--unified`, because it was covered in the examples above (except it's a standard flag which is used every time)
* Since we now have more space we can explain `--new-file` better.
```suggestion
- Create a patch file from the differences of two text files, treating nonexistent files as empty:

`diff --text --new-file {{old_file}} {{new_file}} > {{diff.patch}}`
```

Reply:
I believe if `--unified` is not there, then the format is not a conventional patch file as there will be no context lines (making patching much more risky.
About the change of description, I agree. Is there any character limit per line?

Reply:
Do you think it would help if we write `Create a patch file for Git`? Because that's what `--unified` does right?
Regarding the line lenght. We try to keep it as short as possible, but the above shouldn't be a problem.

Reply:
The limit per line I think is quite high, as the `whiptail` page is still ok.

`Create a patch file for Git` sounds good!

Reply:
So something like this would be satisfactory?

```markdown
Create a patch file for Git
`diff --text --unified --new-file {{old_file}} {{new_file}} > {{diff.patch}}`
```

Reply:
@sbrl why not

> Create a patch file for Git from the differences of two text files, treating nonexistent files as empty

Otherwise we would just throw flags on the user without explaining them

Reply:
if we don't hit the character limit, I agree with @marchersimon